test/parallel/test-repl-persistent-history.js

```diff
@@ -31,21 +30,19 @@ class ActionStream extends stream.Stream {
       const next = _iter.next();
       if (next.done) {
         // Close the repl. Note that it must have a clean prompt to do so.
-        setImmediate(() => {
-          this.emit('keypress', '', { ctrl: true, name: 'd' });
-        });
+        this.emit('keypress', '', { ctrl: true, name: 'd' });
         return;
       }
       const action = next.value;
 
       if (typeof action === 'object') {
         this.emit('keypress', '', action);
       } else {
-        this.emit('data', `${action}\n`);
+        this.emit('data', `${action}`);
```

Comment:
I guess we could directly pass through action? Good find that the \n is obsolete!

```suggestion
        this.emit('data', action);
```